libcontainer/cgroups/fs/memory.go

```diff
@@ -269,3 +284,76 @@ func getMemoryData(path, name string) (cgroups.MemoryData, error) {
 
 	return memoryData, nil
 }
+
+func getPageUsageByNUMA(cgroupPath string) (cgroups.PageUsageByNUMA, error) {
+	stats := cgroups.PageUsageByNUMA{}
+
+	file, err := os.Open(path.Join(cgroupPath, "memory.numa_stat"))
+	if err != nil {
+		return stats, err
+	}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var statsType string
+		statsByType := cgroups.PageStats{Nodes: map[uint8]uint64{}}
+		columns := strings.SplitN(scanner.Text(), numaStatColumnSeparator, numaStatMaxColumns)
+
+		for _, column := range columns {
+			pagesByNode := strings.SplitN(column, numaStatKeyValueSeparator, 2)
+			if len(pagesByNode) != 2 {
+				return cgroups.PageUsageByNUMA{}, fmt.Errorf("wrong data format, found %d fields instead of 2", len(pagesByNode))
+			}
+
+			if isNUMANode.MatchString(pagesByNode[0]) {
```

Comment:
Is this case correct? You are not setting `statsType` there, so this will lead to error when later you will call `addNUMAStatsByType`, or you are sure that this will never be the first case?

Is that covered by test?

Reply:
I think that cgroups in Linux would have to alter their behaviour significantly: at the moment there are only two types of entries per each line in `memory.numa_stat` file.